### PR TITLE
Add alpha exports to notifications and signals

### PR DESCRIPTION
**Pull request by mareklibra**
Adding missing alpha.ts exports to notifications and signals.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mareklibra** [comment]
Cc @freben

**Comment by mareklibra** [comment]
@davidfestal Can you please have a look at this PR?

The point is in being able to export this plugin as a dynamic one via 
`${JANUS_OR_BACKSTAGE_CLI}  package export-dynamic-plugin` without additional code changes.

Side question: Do we need to update `files` section of the `package.json`?

File: plugins/notifications-backend-module-email/src/alpha.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { BackendDynamicPluginInstaller } from '@backstage/backend-dynamic-feature-service';
+import { notificationsModuleEmail } from './module';
+
+/** @alpha */
+export const dynamicPluginInstaller: BackendDynamicPluginInstaller = {
```
Comment (gashcrumb) [inline plugins/notifications-backend-module-email/src/alpha.ts:20]: No problem having this in place for now, but eventually this shouldn't be needed except for very specific circumstances.

**Comment by gashcrumb** [comment]
> The point is in being able to export this plugin as a dynamic one via `${JANUS_OR_BACKSTAGE_CLI} package export-dynamic-plugin` without additional code changes.
> 
> Side question: Do we need to update `files` section of the `package.json`?

This should not be necessary.  If I recall this was previously a requirement because an npm registry was being used to distribute dynamic plugins at runtime, however there's better ways to do this runtime distribution.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Nope sorry, this is not something we're adding to packages. We've settled on the pattern of using the `default` export as they way to export features from packages.

**Pull request closed without merging**